libs/application/ui-fields/src/lib/FindVehicleFormField/FindVehicleFormField.tsx

```diff
@@ -0,0 +1,339 @@
+  const [vehicleDetails, setVehicleDetails] = useState<VehicleDetails | null>(
+    null,
+  )
+  const [submitButtonDisabledCalled, setSubmitButtonDisabledCalled] =
+    useState(false)
+  const updateInputState = (value: string) => {
+    setButtonDisabled(value.length !== 5)
+    setPlate(value)
+  }
+  const findVehicleByPlate = async () => {
+    setIsLoading(true)
+    try {
+      if (!getVehicleDetails) {
+        throw new Error('getVehicleDetails function is not defined')
+      }
+
+      const response = await getVehicleDetails(plate.toUpperCase())
+      const isVehicleFound =
+        isVehicleOperatorChangeChecksByPermno(response) ||
+        isVehicleOwnerchangeChecksByPermno(response) ||
+        isVehiclePlateOrderChecksByPermno(response)
+      setVehicleNotFound(!isVehicleFound)
+
+      if (isVehicleFound) {
+        const vehicleDetails = extractVehicleDetails(response)
+        setVehicleDetails(vehicleDetails)
+        setPlate(plate)
+        setValue(`${field.id}.type`, vehicleDetails.make)
+        setValue(`${field.id}.make`, vehicleDetails.make)
+        setValue(`${field.id}.plate`, plate)
+        setValue(`${field.id}.color`, vehicleDetails.color || undefined)
+        setValue(`${field.id}.requireMilage`, vehicleDetails.requireMilage)
+        setValue('vehicleInfo.plate', plate)
+        setValue('vehicleInfo.type', vehicleDetails.make)
+        setSubmitButtonDisabled && setSubmitButtonDisabled(false)
+      } else {
+        setVehicleNotFound(true)
+        setSubmitButtonDisabled && setSubmitButtonDisabled(true)
+      }
+    } catch (error) {
+      console.error('error', error)
+      setVehicleNotFound(true)
+      setVehicleDetails(null)
+      setSubmitButtonDisabled && setSubmitButtonDisabled(true)
+    } finally {
+      setIsLoading(false)
+    }
+  }
+  const isDisabled =
+    additionalErrors &&
+    vehicleDetails &&
+    (!vehicleDetails.isDebtLess ||
+      !!vehicleDetails.validationErrorMessages?.length)
+
+  useEffect(() => {
+    if (!submitButtonDisabledCalled) {
+      setSubmitButtonDisabled && setSubmitButtonDisabled(true)
+      setSubmitButtonDisabledCalled(true)
+    }
+    if (plate.length === 5) {
+      setButtonDisabled(false)
+    }
+    setFieldLoadingState?.(isLoading)
+  }, [isLoading])
+
+  return (
+    <Box>
+      <Box display="flex" alignItems="center">
+        <Box marginRight={2}>
+          <InputController
+            id="findVehicle.plate"
+            name="findVehicle.plate"
+            label={
+              findPlatePlaceholder &&
+              formatText(findPlatePlaceholder, application, formatMessage)
+            }
+            onChange={(event) => {
+              updateInputState(event.target.value)
+            }}
+            required={true}
+            defaultValue={plate}
+            rules={{
+              required: true,
+              validate: (value) => {
+                if (value.length !== 5) {
```

Comment:
If introducing magic numbers like this they should be in a constant at the top to improve readability or a prop.

Reply:
Fixed